formats/mpq/mpq.go

```diff
@@ -65,7 +72,7 @@ func (f *file) Read(data []byte) (int, error) {
 	var bytesRead int
 	if result := C.SFileReadFile(f.handle, unsafe.Pointer(&data[0]), C.uint(len(data)), (*C.uint)(unsafe.Pointer(&bytesRead)), nil); result == 0 {
 		lastError := getLastError()
-		if lastError == 1002 { // ERROR_HANDLE_EOF
+		if lastError == 1002 || lastError == 38 { // ERROR_HANDLE_EOF
```

Comment:
This needs to be conditional based on platform. Who knows what error `38` is on Linux and error `1002` on Windows?

Reply:
of course, i was just making the testing/tool/sample work, not correct :)   But i'll make this more elegant just for you

Reply:
Added platform dependent "sysEOF" constant